teuthology/task/install.py

```diff
@@ -927,6 +926,121 @@ def _upgrade_deb_packages(ctx, config, remote, debs):
     )
 
 
+@contextlib.contextmanager
+def rh_install(ctx, config):
+    """
+    Installs rh ceph on all hosts in ctx.
+
+    :param ctx: the argparse.Namespace object
+    :param config: the config dict
+    """
+    version = config['rhbuild']
+    rh_versions = ['1.3.0']
+    if version in rh_versions:
+        log.info("Supported version")
+    else:
+        raise RuntimeError("Unsupported RH Ceph version %s " , version)
+
+    with parallel() as p:
+        for remote in ctx.cluster.remotes.iterkeys():
+            if remote.os.name == 'rhel':
+                log.info("Installing on RHEL node")
+                p.spawn(rh_install_pkgs, ctx, remote, version)
+            else:
+                raise RuntimeError("Test requires RHEL nodes")
+    try:
+        yield
+    finally:
+        if config.get('skip_uninstall'):
+            log.info("Skipping uninstall of Ceph")
+        else:
+            rh_uninstall(ctx=ctx, config=config)
+
+
+def rh_uninstall(ctx, config):
+    """
+     Uninstalls rh ceph on all hosts.
+     It actually spawns rh_uninstall_pkgs() on the remotes for uninstall.
+
+    :param ctx: the argparse.Namespace object
+    :param config: the config dict
+    """
+    with parallel() as p:
+        for remote in ctx.cluster.remotes.iterkeys():
+            p.spawn(rh_uninstall_pkgs, ctx, remote)
+
+
+def rh_install_pkgs(ctx, remote, installed_version):
+    """
+    Installs RH build using ceph-deploy.
+
+    :param ctx: the argparse.Namespace object
+    :param remote: the teuthology.orchestra.remote.Remote object
+    """
+    pkgs = ['ceph-deploy']
+    rh_version_check = {'0.94.1':'1.3.0'}
+    for pkg in pkgs:
+        log.info("Check if ceph-deploy is already installed on system")
```

Comment:
Since this function will be run in parallel on multiple hosts, all the log messages should mention the hostname (``remote.shortname`` works well here), or else they won't be helpful at all